aggoracle/chaingersender/evm.go

```diff
@@ -41,48 +64,92 @@ type EVMChainGERSender struct {
 	}
 
-	return &EVMChainGERSender{
-		logger:              logger,
-		l2GERManager:        l2GERManager,
-		l2GERManagerAddr:    l2GERManagerAddr,
-		l2GERManagerAbi:     l2GERAbi,
-		ethTxMan:            ethTxMan,
-		gasOffset:           gasOffset,
-		waitPeriodMonitorTx: waitPeriodMonitorTx,
-	}, nil
+	sender := &EVMChainGERSender{
+		logger:                 logger,
+		mode:                   mode,
+		l2GERManager:           l2GERManager,
+		l2GERManagerAddr:       l2GERManagerAddr,
+		l2GERManagerAbi:        l2GERAbi,
+		aggOracleCommitteeAddr: aggOracleCommitteeAddr,
+		l2Client:               l2Client,
+		ethTxMan:               ethTxMan,
+		gasOffset:              gasOffset,
+		waitPeriodMonitorTx:    waitPeriodMonitorTx,
+	}
+
+	// Initialize mode-specific components
+	if err := sender.initializeMode(); err != nil {
+		return nil, err
+	}
+
+	logger.Infof("EVMChainGERSender initialized in %s mode", mode)
+	return sender, nil
 }
 
-// validateGERSender validates whether the provided GER sender is allowed to send and remove GERs
-func validateGERSender(gerSender common.Address, l2GERManagerSC types.L2GERManagerContract) error {
-	zeroAddr := common.Address{}
-	gerUpdater, err := l2GERManagerSC.GlobalExitRootUpdater(nil)
+// initializeMode initializes mode-specific components and validations
+func (c *EVMChainGERSender) initializeMode() error {
+	switch c.mode {
+	case DirectInjectionMode:
+		return c.initializeDirectInjectionMode()
+	case AggOracleCommitteeMode:
+		return c.initializeAggOracleCommitteeMode()
+	default:
+		return fmt.Errorf("unknown GER mode: %s", c.mode)
+	}
+}
+
+func (c *EVMChainGERSender) initializeDirectInjectionMode() error {
+	if err := validateGERSender(c.ethTxMan.From(), c.l2GERManager); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *EVMChainGERSender) initializeAggOracleCommitteeMode() error {
+	// Create AggOracleCommittee contract binding
+	aggOracleCommittee, err := aggoraclecommittee.NewAggoraclecommittee(c.aggOracleCommitteeAddr, c.l2Client)
 	if err != nil {
-		return fmt.Errorf("failed to retrieve GER updater address from GER L2 manager: %w", err)
+		return fmt.Errorf("failed to create binding for AggOracleCommittee (SC address: %s): %w",
+			c.aggOracleCommitteeAddr, err)
 	}
 
-	if gerUpdater != zeroAddr && gerSender != gerUpdater {
-		return fmt.Errorf("invalid GER sender provided (in the EthTxManager configuration), "+
-			"and it is not allowed to update GERs. Expected GER updater by the L2 GER manager contract: %s", gerUpdater)
+	// Get the ABI for AggOracleCommittee
+	aggOracleCommitteeAbi, err := aggoraclecommittee.AggoraclecommitteeMetaData.GetAbi()
+	if err != nil {
+		return fmt.Errorf("failed to retrieve AggOracleCommittee ABI: %w", err)
+	}
+
+	c.aggOracleCommittee = aggOracleCommittee
+	c.aggOracleCommitteeAbi = aggOracleCommitteeAbi
+
+	// Validate GER proposer
+	if err := validateGERProposer(c.ethTxMan.From(), aggOracleCommittee); err != nil {
+		return err
 	}
 
 	return nil
```

Comment:
Let's validate early and if we detect any issues, it is not necessary to fetch the abi and set the fields, WDYT?

```suggestion
	aggOracleCommittee, err := aggoraclecommittee.NewAggoraclecommittee(c.aggOracleCommitteeAddr, c.l2Client)
	if err != nil {
		return fmt.Errorf("failed to create binding for AggOracleCommittee (SC address: %s): %w",
			c.aggOracleCommitteeAddr, err)
	}

    // Validate GER proposer
	if err := validateGERProposer(c.ethTxMan.From(), aggOracleCommittee); err != nil {
		return err
	}

	// Get the ABI for AggOracleCommittee
	aggOracleCommitteeAbi, err := aggoraclecommittee.AggoraclecommitteeMetaData.GetAbi()
	if err != nil {
		return fmt.Errorf("failed to retrieve AggOracleCommittee ABI: %w", err)
	}

	c.aggOracleCommittee = aggOracleCommittee
	c.aggOracleCommitteeAbi = aggOracleCommitteeAbi

	return nil
```

Reply:
done